google/cloud/storage/examples/storage_async_samples.cc

```diff
@@ -140,6 +140,40 @@ void InsertObjectVectorVectors(
   (client, argv.at(0), argv.at(1));
 }
 
+#if GOOGLE_CLOUD_CPP_HAVE_COROUTINES
+void ReadObject(google::cloud::storage_experimental::AsyncClient& client,
+                std::vector<std::string> const& argv) {
+  //! [read-object]
+  namespace gcs_ex = google::cloud::storage_experimental;
+  auto coro =
+      [](gcs_ex::AsyncClient& client, std::string bucket_name,
+         std::string object_name) -> google::cloud::future<std::uint64_t> {
+    auto [reader, token] = (co_await client.ReadObject(std::move(bucket_name),
+                                                       std::move(object_name)))
+                               .value();
+    std::uint64_t count = 0;
+    while (token.valid()) {
+      auto [payload, t_] = (co_await reader.Read(std::move(token))).value();
```

Comment:
Why `t_`?

Reply:
AFAIK `auto [payload, token]` would introduce a "new" `token`, and would not affect the value of `token` in the `while()` loop.

Reply:
Oh, maybe you are asking "why the underscore"?  As opposed to a simple name, no reason. Changed.